### PR TITLE
Switch to sbt 1.0.0-M6 (WIP)

### PR DESCRIPTION
**Pull request by alexarchambault**

### CODE REVIEW

File: project/Settings.scala [hunk 1]
```diff
@@ -178,11 +178,91 @@ object Settings {
...
+      }
+    }.evaluated
+  }
+
   lazy val plugin =
     javaScalaPluginShared ++
     divertThingsPlugin ++
     withScriptedTests ++
     Seq(
+      sbtLauncher := {
```
Comment (jvican) [inline project/Settings.scala]: Why is this required?

File: sbt-coursier/src/main/scala/coursier/Tasks.scala [hunk 2]
```diff
@@ -861,19 +863,24 @@ object Tasks {
 
       val verbosityLevel = coursierVerbosity.value
 
+      val classifiersRes = coursierSbtClassifiersResolution.value
+      val mainRes = coursierResolutions.value
+
       val res =
         if (withClassifiers && sbtClassifiers)
-          Seq(coursierSbtClassifiersResolution.value)
```
Comment (jvican) [inline sbt-coursier/src/main/scala/coursier/Tasks.scala]: :smile:

File: project/Settings.scala (hunk 1 above)
Comment (alexarchambault) [inline project/Settings.scala]: The [default `sbtLauncher` implem](https://github.com/sbt/sbt/blob/6ce09cfed6f3353326981a4e6e07ffba1f15252f/scripted/plugin/src/main/scala/sbt/ScriptedPlugin.scala#L102) expects the sbt launcher to appear first in the update report.  But coursier currently gives no guaranties about the order of the modules and artifacts in its output, so not in the update report either. For whatever reasons, it was in first position anyway before, but isn't anymore now...

File: sbt-coursier/src/main/scala/coursier/Tasks.scala (hunk 2 above)
Comment (alexarchambault) [inline sbt-coursier/src/main/scala/coursier/Tasks.scala]: Your fault 😛 But yeah, it's better this way overall.

File: project/Settings.scala (hunk 1 above)
Comment (jvican) [inline project/Settings.scala]: Hm, I wonder why it's not the first anymore. Thanks for explaining!

**Comment by alexarchambault** [comment]
(Failing test on Travis is fine locally and on Windows CI.)
